File: app/src/main/java/com/sxshi/retrofit/utils/ConnectivityUtils.java
package com.sxshi.retrofit.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * 网络检测工具，需要更新
 */
public final class ConnectivityUtils {

    private ConnectivityUtils() {
    }

    public static boolean isConnectedToInternet(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        return ni != null && ni.isConnected() && ni.isAvailable();
    }

    public static boolean isWifiEnabled(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        return ni != null && ni.getType() == ConnectivityManager.TYPE_WIFI;
    }

    public static boolean isMobileDataEnabled(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        return ni != null && ni.getType() == ConnectivityManager.TYPE_MOBILE;
    }

}

File: app/src/main/java/com/sxshi/retrofit/utils/ConnectivityUtils.kt
package com.sxshi.retrofit.utils

import android.content.Context
import android.net.ConnectivityManager

/**
 * 网络检测工具，需要更新
 */
object ConnectivityUtils {
    fun isConnectedToInternet(context: Context): Boolean {
        val cm = context
                .applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val ni = cm.activeNetworkInfo
        return ni != null && ni.isConnected && ni.isAvailable
    }

    fun isWifiEnabled(context: Context): Boolean {
        val cm = context
                .applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val ni = cm.activeNetworkInfo
        return ni != null && ni.type == ConnectivityManager.TYPE_WIFI
    }

    fun isMobileDataEnabled(context: Context): Boolean {
        val cm = context
                .applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val ni = cm.activeNetworkInfo
        return ni != null && ni.type == ConnectivityManager.TYPE_MOBILE
    }
}